### PR TITLE
Add id field to Book type

### PR DESCRIPTION
**Pull request by SilverMaiden**
What it says on the tin!

### CODE REVIEW

File: src-ui/routes/+page.svelte [hunk 1]
```diff
@@ -43,10 +45,11 @@
     return {
       destroy: (): void => {
         onScan.detachFrom(node)
-      }
+      },
     }
   }
 </script>
+
```
Comment (SilverMaiden) [inline src-ui/routes/+page.svelte]: Prettier added these on save

File: src-ui/lib/types/book.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export type Book = {
+  id: number
```
Comment (rkofman) [inline src-ui/lib/types/book.ts]: Let's make `id` a string. We'll generally want UUIDs. Unless you're intending to display it; in which case we can leave it as a human-readable format. Likely still string based though.

File: src-ui/routes/+page.svelte [hunk 3]
```diff
@@ -8,12 +8,14 @@
   import { getByISBN } from '../lib/openLibrary.js'
   import { books } from '../lib/state/Books.svelte'
 
+  let idCounter = 1
```
Comment (rkofman) [inline src-ui/routes/+page.svelte]: this shouldn't live on the page. Any ideas where would be a better spot?

File: package.json [hunk 4]
```diff
@@ -30,7 +30,8 @@
     "bulma": "^1.0.2",
     "onscan.js": "^1.5.2",
     "open-library-api": "^0.1.2",
-    "openapi-fetch": "^0.12.2"
+    "openapi-fetch": "^0.12.2",
+    "uuid": "^10.0.0"
```
Comment (rkofman) [inline package.json:34]: might be possible to remove in favor of https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

I'm not :100: sure, but would be neat if true.

File: src-ui/lib/state/Books.svelte.ts [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import { type Book } from '../types/book.js'
+import { v4 as uuidv4 } from 'uuid'
 
 type BooksStore = {
   value: Book[]
-  add: (book: Book) => void
+  add: (book: Book | Partial<Book>) => void
```
Comment (rkofman) [inline src-ui/lib/state/Books.svelte.ts]: A book is already a Partial book, so you might be able to just make the type be the partial.

Comment (rkofman) [reply src-ui/lib/state/Books.svelte.ts]: or you might be able to get even more specific by declaring this to expect a Book minus the ID field plus an optional ID field. Another alternative: create a new type of `UnsavedBook` that doesn't have an ID, but has all other fields.

**Comment by rkofman** [comment]
Great except for type wrangling and possible library removal -- which can both be follow-up PRs, tbh.

**Review by rkofman — APPROVED** [review]

File: src-ui/lib/components/BooksTable.svelte [hunk 6]
```diff
@@ -19,6 +17,7 @@
 
   if (books.value.length == 0) {
     books.add(initialBook)
+    console.log(books.value)
```
Comment (rkofman) [inline src-ui/lib/components/BooksTable.svelte]: console.log

File: src-ui/lib/state/Books.svelte.ts [hunk 7]
```diff
@@ -22,6 +22,8 @@ function createBooks(): BooksStore {
       val = newVal
     },
     add: (book: Book | Partial<Book>): void => {
+      // NOTE (isummit): we're using this instead of window.crypto because window.crypto.randomUUID is
+      // not available for older versions of mac os, which some of our users may be on.
```
Comment (rkofman) [inline src-ui/lib/state/Books.svelte.ts]: Can we note _which_ versions?

**Comment by rkofman** [comment]
@SilverMaiden ready to merge yet?

**Pull request merged**